src/main/java/taxi/controller/driver/LoginController.java

```diff
@@ -0,0 +1,42 @@
+package taxi.controller.driver;
+
+import java.io.IOException;
+import javax.security.sasl.AuthenticationException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import taxi.lib.Injector;
+import taxi.model.Driver;
+import taxi.service.AuthenticationService;
+
+public class LoginController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("taxi");
+    private AuthenticationService authService;
+
+    @Override
+    public void init() {
+        authService =
+                (AuthenticationService) injector.getInstance(AuthenticationService.class);
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/drivers/login.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        String login = req.getParameter("login");
+        String password = req.getParameter("password");
+        try {
+            Driver driver = authService.login(login, password);
+            req.getSession().setAttribute("driver_id", driver.getId());
+            resp.sendRedirect(req.getContextPath() + "/index");
+        } catch (AuthenticationException e) {
+            throw new RuntimeException("invalid login or password, try again");
```

Comment:
```suggestion
            throw new RuntimeException(e.getMessage());
```
If we are already throwing an exception with a custom message, we can reuse it this way.